.github/workflows/scripts/common.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# The output appears in a dedicated emphasized GitHub step box.
+# Multiple lines are not supported.
+# Markdown is not supported.
+gh_log() {
+    local LEVEL="$1"
+    shift
+    echo "::$LEVEL::$*"
+}
+export -f gh_log
+
+# Sets a step output value.
+# Multiple lines are not supported: join with URL escaping.
+gh_output() {
+    local NAME="$1"
+    shift
+    echo "::set-output name=$NAME::$*"
+}
+export -f gh_output
+
+check_not_empty() {
+    for V in "$@"; do
+        typeset -n VAR="$V"
+        if [ -z "${VAR:-}" ]; then
+            gh_log error "Variable $V is not set or empty"
+            exit 1
+        fi
+    done
+}
+export -f check_not_empty
+
+check_not_empty GITHUB_STEP_SUMMARY
+
+# The output appears in the GitHub step summary box.
+# Multiple lines are supported.
+# Markdown is supported.
+gh_summary() {
+    if [ "$#" -eq 0 ]; then
+        cat
+    else
+        echo -e "$@"
+    fi >>"$GITHUB_STEP_SUMMARY"
+}
+export -f gh_summary
+
+if [ "$#" -gt 0 ]; then
+    SCRIPT="$1"
+    check_not_empty \
+        SCRIPT \
+        GITHUB_REPOSITORY \
+        main_branch
+    URL="/repos/$GITHUB_REPOSITORY/contents/.github/workflows/scripts/$SCRIPT.sh?ref=$main_branch"
+    shift
+    gh api -H "Accept: application/vnd.github.v3.raw" "$URL" | bash -s -- "$@"
+fi
```

Comment:
I see now :) 
[This construct](https://github.com/stackrox/stackrox/blob/master/release/scripts/scan-images-with-roxctl.sh#L60) allows to control the behavior of the script when running it and when sourcing it. 

Just incase you want to add extra protection here (or run a command that accepts no parameters).

Reply:
Thanks. I don't want extra protection here :)